src/components/Activities/Activity3/Game/TimeAllocation.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useContext, useEffect} from 'react';
+import { GameContext } from '.';
+
+import Clock from '../../../../assets/activity3/game/Clock.svg';
+import Debug from '../../../../assets/activity3/game/Debug.svg';
+import Graph from '../../../../assets/activity3/game/Graph.svg';
+import Hammer from '../../../../assets/activity3/game/Hammer.svg';
+
+import { STARTING_DAYS, LOW_DAY_THRESHOLD, HIGH_DAY_THRESHOLD } from './GameConstants';
+import NumberSelection from './NumberSelection';
+
+interface TimeAllocationProps {
+  setTimeAllocation: (variables: number[]) => void;
+  initialTimes: number[];
+}
+
+function TimeAllocation(props: TimeAllocationProps): JSX.Element {
+  const {daysLeft, setDaysLeft, goNextState} = useContext(GameContext);
+  const { initialTimes, setTimeAllocation } = props;
+  const [daysAllocation, setDaysAllocation] = useState<number[]>([0,0,0]);
+
+  const DISPLAY_OPTIONS = [
+    { src: Hammer, text: 'Build' },
+    { src: Debug, text: 'Debug' },
+    { src: Graph, text: 'A/B Test' },
+  ];
+
+  const displayOptionIcons = () => {
+    return DISPLAY_OPTIONS.map((option) => (
+      <div key={option.text} className={'centered-box'}>
+        <img src={option.src}/>
+        {option.text}
+      </div>
+    ));
+  };
+
+  useEffect(() => {
+    // reallocate task distribution
+    const initAllocation = initialTimes ?? [0,0,0];
+    setDaysAllocation(initAllocation);
+
+    // reset daysLeft to maximum
+    setDaysLeft && setDaysLeft(STARTING_DAYS);
+  }, []);
+
+  const handleGoNext = () => {
+    // update the new task distribution
+    setTimeAllocation(daysAllocation);
+
+    // update the number of remaining days
+    setDaysLeft && setDaysLeft(daysLeft ? (daysLeft - sumDaysUsed()) : 0);
+    goNextState && goNextState();
+  };
+
+  const sumDaysUsed = () : number => {
+    return daysAllocation.reduce((acc : number, cur : number) => acc + cur);
+  };
+
+  const displayWarning = () => {
```

Comment:
more clear that this return JSX Element, and that it doesn't cause a display of warning
```suggestion
  const getDisplayWarning = (): JSX.Element => {
```